### DISCUSSION TITLE
How to get all custom tasks ？

### DISCUSSION THREAD

**Discussion by Iwith** [celery/django-celery-beat#461] [Q&A]
### Summary:

Include a *brief* descrioption of the problem here, and fill out the version info below.  

* Celery Version: celery==4.4.2
* Celery-Beat Version: django-celery-beat==2.2.0

### Exact steps to reproduce the issue:
1. I define tasks in the task. py file in each application directory
2. after app.autodiscover_tasks()
3. I can not get custom task in views.py

### Detailed information

`
# app01.views.py
from django.http import HttpResponse,JsonResponse
from dj_celery_pro import celery_app
class Task_test(View):

    def get(self, request):
        all_task =list(name for name in celery_app.tasks if not name.startswith('celery.'))   # It can not get all custom tasks in tasks.py of each application
        return JsonResponse(dict(data=all_task))
`

When I access the view-class, it only returned '{"data": ["dj_celery_pro.celery.debug_task"]}' but no contain custom tasks.

Can anyone help me?  Thanks.

**Comment by nick2wang** [comment]
This works for me.
```
from celery import current_app

i = current_app.control.inspect()
tasks = i.registered_tasks()
```
